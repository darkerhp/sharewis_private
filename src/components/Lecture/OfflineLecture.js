import React, { PropTypes } from 'react';
import ReactNative from 'react-native';
import I18n from 'react-native-i18n';

const { View, StyleSheet, Text } = ReactNative;


const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  contentText: {
    textAlign: 'center',
    lineHeight: 22,
    fontSize: 17,
    color: '#222',
  },
});


const OfflineLecture = ({ lectureContentStyleId }) =>
  (
    <View style={[lectureContentStyleId, styles.container]}>
      <View>
        <Text style={styles.contentText}>
          {I18n.t('offlineLecture')}
        </Text>
      </View>
    </View>
  );

OfflineLecture.propTypes = {
  lectureContentStyleId: PropTypes.number.isRequired,
};

export default OfflineLecture;
