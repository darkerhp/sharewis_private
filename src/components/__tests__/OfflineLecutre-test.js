/* eslint-disable no-undef */
/* eslint-disable import/no-extraneous-dependencies */
import 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import OfflineLecture from '../Lecture/OfflineLecture';

jest.mock('react-native-i18n', () => ({ t: () => '' }));

describe('OfflineLecture', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<OfflineLecture lectureContentStyleId={1} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
