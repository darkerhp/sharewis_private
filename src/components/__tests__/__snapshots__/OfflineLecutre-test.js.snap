// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfflineLecture renders correctly 1`] = `
<View
  style={
    Array [
      1,
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#222",
          "fontSize": 17,
          "lineHeight": 22,
          "textAlign": "center",
        }
      }
    >
      
    </Text>
  </View>
</View>
`;
